fluentd/run.sh

```diff
@@ -129,32 +125,55 @@ else
 fi
 export K8S_FILTER_REMOVE_KEYS
 
-if [ "${USE_MUX:-}" = "true" ] ; then
-    TOTAL_MEMORY_LIMIT=`echo $MUX_MEMORY_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc`
+# How many outputs?
+if [ -n "${MUX_CLIENT_MODE:-}" -o "${USE_MUX_CLIENT:-}" = "true" ] ; then
+    # A fluentd collector configured as a mux client has just one output: sending to a mux.
+    NUM_OUTPUTS=1
 else
-    TOTAL_MEMORY_LIMIT=`echo $FLUENTD_MEMORY_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc`
+    # fluentd usually has 2 outputs.
+    NUM_OUTPUTS=2
+    if [ "$ES_COPY" = "true" ]; then
+        NUM_OUTPUTS=`expr $NUM_OUTPUTS \* 2`
+    fi
 fi
-BUFFER_SIZE_LIMIT=`echo $BUFFER_SIZE_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc`
-if [ $BUFFER_SIZE_LIMIT -eq 0 ]; then
-    BUFFER_SIZE_LIMIT=1048576
+
+FILE_BUFFER_PATH=/var/lib/fluentd
+if [ ! -d $FILE_BUFFER_PATH ]; then
+    rm -rf $FILE_BUFFER_PATH || :
+    mkdir -p $FILE_BUFFER_PATH
```

Comment:
I think I haven't understood your concern yet...

I thought if a non-directory /var/lib/fluentd was found, it'd be better to be replaced with /var/lib/fluentd directory.  I guess this is what you disagree?

If we remove the line 141 check and always call mkdir, it'd fail as you commented in your original comment (if non-directory /var/libfluentd is there).  That is, fluentd/run.sh is supposed to fail there instead of going forward?  If so, do we want to add an error message about it?  Or the error message from the mkdir command is good enough?